klaviyo docs onboarding, minor doc fix for AC (#27)
rudderlabs/rudderstack-docs: destinations/klaviyo.md

The `track` call allows you to capture any action that the user might perform, and the properties associated with that action. Each action is considered to be an event. It is similar to `screen` event, and the user has to be associated with an event either by using `email` or `phone`.

A sample `track` call looks like the following:

```javascript
rudderanalytics.track(
  "Checked Out",
  {
    Clicked_Rush_delivery_Button: true,
    total_value: 2000,
    Odered: ["T-Shirt", "jacket"],
  },
  {
    traits: {}
      email: "name@website.com",
      phone: "123456789"
    },
  }
);
```

In the above snippet, RudderStack captures the information related to the `Checked Out` event, along with any additional info about that event - in this case the name of the product.

{% hint style="info" %}
If you want to set a specific value to the `screen` and `track` type event, you need to pass the `event` related property in event properties. For the user-related information, you need to pass them in `traits` in `context`, as shown in the example above.
{% endhint %}

{% hint style="info" %}
Apart from either `email` or `phone`, the other fields are not mandatory.
{% endhint %}

## Identify

> The `identify` call lets you associate a user with their actions and capture all the relevant traits about them. This information includes unique `userid` as well as any optional information such as name, email address, etc.

Reviewer: Please review this. Is email optional or mandatory? In some of the other sections, we mentioned that `name` and `email` are required. We say:

"The `email` or `phone` trait is a mandatory trait for mapping a user to Klaviyo."

@utsabc @arnab-p

Author: Sorry I missed this part, I Klaviyo we are mapping the user unique to email or phone, i.e either of these two is required to make an identity call. So here `userid` is not mandatory or uniquely used in this destination. Other `traits` are optional.

To send `name` and `category` info in the page event, you can do so by switching on the `Send Page As Track` button on the RudderStack dashboard. And, to associate `properties` with the page-view event, you can do so by enabling the `Additional Page info` property on the RudderStack dashboard.

An example of such `page` call is shown below.

```javascript
rudderanalytics.page("Cart", "Cart Viewed", {
  path: "/cart",
  referrer: "test.com",
  search: "term",
  title: "test_item",
  url: "http://test.in",
});
```

{% hint style="info" %}
Note that **`page` calls are only supported in the RudderStack device mode** integration. To know more about the difference between Cloud mode and Device mode in RudderStack, read the [RudderStack connection modes](https://docs.rudderstack.com/get-started/rudderstack-connection-modes) guide.
{% endhint %}

## Screen

The `screen` method allows you to record whenever a user sees the mobile screen, along with any associated optional properties. This call is similar to the `page` call, but is exclusive to your mobile device.

A sample `screen` call looks like the following code snippet:

> ```javascript

Reviewer 2: ```suggestion
```xcode
```

{% endhint %}

{% hint style="info" %}
Similar to `listId` adding `consent` and `smsConsent` property will override the value stored in the control plane for the specific event.
{% endhint %}

{% hint style="info" %}
Adding or subscribing users to specific list is only available in cloud-mode integration. To know more about the difference between Cloud mode and Device mode in RudderStack, read the [RudderStack connection modes](https://docs.rudderstack.com/get-started/rudderstack-connection-modes) guide.
{% endhint %}

## Group

The `group` call lets you associate a user with a list. You can also subscribe a user to the list. If you haven't already added or subscribed the user using the `identify` call, you can do so by using `group` call.

A sample `group` call looks like the following:

```javascript

> rudderanalytics.group(

Reviewer 3: use node js group call example

Author: done

- **Consent:** If you are a GDPR-compliant business, you will need to include `consent` in your API call.`consent` is a Klaviyo-specific property and only accepts the following values: `email`, `web`, `sms`, `directmail`, and `mobile`.
- **SMS Consent:** If you are updating the consent for a phone number, or would like to send an opt-in SMS to the profile (for double opt-in lists), include an `smsConsent` key in the properties with a value of `true` or `false`.

## Page

The `page` call allows you the web page that a user is viewing, along with its associated properties.

An example `page` call would look like:

```javascript
rudderanalytics.page();
```

If you want to send additional info in the `page` call, you can do it so by adding the `additionalInfo` key in the page event properties along with other `pageInfo` objects that you want to associate with the `page` event. 

Here is an example:

```javascript
rudderanalytics.page(
  "Cart",
  "Cart Viewed",
  {
    additionalInfo: true,
    pageInfo: {
      path: "/cart",
      referrer: "test.com",
      search: "term",
      title: "test_item",
      url: "http://test.in",
    }
  },
  () => {
    console.log("in page call");

>   }

Reviewer 2: Let's remove this part

Author: done

{% endhint %}

{% hint style="info" %}
Similar to `listId` adding `consent` and `smsConsent` property will override the value stored in the control plane for the specific event.
{% endhint %}

{% hint style="info" %}
Adding or subscribing users to specific list is only available in cloud-mode integration. To know more about the difference between Cloud mode and Device mode in RudderStack, read the [RudderStack connection modes](https://docs.rudderstack.com/get-started/rudderstack-connection-modes) guide.
{% endhint %}

## Group

The `group` call lets you associate a user with a list. You can also subscribe a user to the list. If you haven't already added or subscribed the user using the `identify` call, you can do so by using `group` call.

A sample `group` call looks like the following:

```javascript
rudderanalytics.group(

>   "XUepkK",

Reviewer 2: Let's use a value like `sampleGroupId`. It's more intuitive to understand.

Author: done

{% hint style="info" %}
If you want to set a specific value to the `screen` and `track` type event, you need to pass the `event` related property in event properties, aslo you can send `revenue` property in track we will map it to klaviyo special property `$value`.
{% endhint %}

## Identify

The `identify` call lets you associate a user with their actions and capture all the relevant traits about them. This information includes unique `userid` as well as any optional information such as name, email address, etc.

A sample `identify` call looks like the following:

```javascript
rudderanalytics.identify("userid", {
  firstName: "Name",
  lastName: "Surname",
  email: "name@website.com",
  phone: "+12 345 678 900",
  userId: "userId",
  title: "Owner",
  organization: "Company",
  city: "Tokyo",
  region: "ABC",
  country: "JP",
  zip: "100-0001",
  Flagged: false,
  Residence: "Shibuya",
  properties: {
    listId: "XUepkK",
    subscribe: true,
    consent: "email",
    smsConsent: true,
  },
});
```

In the above snippet, RudderStack captures relevant information about the user such as the `email`, `phone` as well as the associated traits of that user.

{% hint style="info" %}

> The `userId`,`email` or `phone` trait is a mandatory trait for mapping a user to Klaviyo. If a user already exists, the new values will be updated for that user . You can further add the user to the list if default `listId` is present in Rudderstack control-plane  or by adding `listId` in the `properties` within the `traits`, this will override the `listId` you used in the control plane for this event. You can also subscribe the user to a list by setting `subscribe` option to `true`.

Reviewer 2: ```suggestion
The `userId`, `email`, or `phone` trait is a mandatory trait for mapping a user to Klaviyo. If a user already exists, the new values will be updated for that user. You can further add the user to the list if default `listId` is present on the RudderStack dashboard. You can also do so by adding `listId` in the `properties` within the `traits`, this will override the `listId` you used on the dashboard for that particular event. You can also subscribe the user to a list by setting `subscribe` option to `true`.
```

{% endhint %}

{% hint style="info" %}

> Similar to `listId` adding `consent` and `smsConsent` property will override the value stored in the control plane for the specific event.

Reviewer 2: ```suggestion
Similar to `listId` adding `consent` and `smsConsent` property will override the settings on the RudderStack dashboard for the specific event.
```

{% endhint %}

{% hint style="info" %}

> Adding or subscribing users to specific list is only available in cloud-mode integration. To know more about the difference between Cloud mode and Device mode in RudderStack, read the [RudderStack connection modes](https://docs.rudderstack.com/get-started/rudderstack-connection-modes) guide.

Reviewer 2: ```suggestion
**Adding or subscribing users to a specific list is only available in cloud-mode** integration. To know more about the difference between Cloud mode and Device mode in RudderStack, read the [RudderStack connection modes](https://docs.rudderstack.com/get-started/rudderstack-connection-modes) guide.
```

{% endhint %}

## Group

The `group` call lets you associate a user with a list. You can also subscribe a user to the list. If you haven't already added or subscribed the user using the `identify` call, you can do so by using `group` call.

A sample server-side `group` call looks like the following:

```javascript
client.group({
  userId: 'userId',
  groupId: 'listId',
  traits: {
    subscribe: true
  },
  context:{
    traits: {
      email: "user.test@gmail.com",
      city: "city",
      country: "country",
      zip: "213456",
      age: 23,
      plan: "free",
      consent: "email",
      phone: '1-617-555-1333',
      smsConsent: true
    }
  }
});
```

In the above snippet, the user with the associated traits is added to list, and also subscribed using the `subscribe` flag.

{% hint style="info" %}

> Apart from either `userId`,`email` or `phone`, the other fields are not mandatory.

Reviewer 2: ```suggestion
Apart from either `userId`, `email` or `phone`, the other fields are not mandatory.
```

{% endhint %}

{% hint style="info" %}

> Adding `consent` and `smsConsent` property in user traits will override the value stored in control plane for the specific event.

Reviewer 2: ```suggestion
Adding `consent` and `smsConsent` property in user traits will override the settings on the RudderStack dashboard for the specific event.
```
